File: app/src/main/java/com/gamecenter/app/SaveManager.java
package com.gamecenter.app;

import android.content.Context;
import android.content.SharedPreferences;

/**
 * 游戏存档管理器（Java 版，用于子模块编译引用）。
 * 运行时由 Kotlin 版 SaveManager 提供实际实现（通过 Hilt 注入或静态单例）。
 */
public final class SaveManager {
    private static final String PREFS_NAME = "GameMatrix_saves";
    private static final String KEY_PREFIX_SAVE = "save_";
    private static final String KEY_PREFIX_PROGRESS = "progress_";

    private static volatile SaveManager instance;
    private final SharedPreferences prefs;

    private SaveManager(Context context) {
        prefs = context.getApplicationContext().getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE);
    }

    public static SaveManager getInstance(Context context) {
        if (instance == null) {
            synchronized (SaveManager.class) {
                if (instance == null) {
                    instance = new SaveManager(context.getApplicationContext());
                }
            }
        }
        return instance;
    }

    public void save(String gameId, String slotKey, String jsonState) {
        prefs.edit().putString(KEY_PREFIX_SAVE + gameId + "_" + slotKey, jsonState).apply();
    }

    public String load(String gameId, String slotKey) {
        return prefs.getString(KEY_PREFIX_SAVE + gameId + "_" + slotKey, null);
    }

    public boolean hasSave(String gameId, String slotKey) {
        return prefs.contains(KEY_PREFIX_SAVE + gameId + "_" + slotKey);
    }

    public void deleteSave(String gameId, String slotKey) {
        prefs.edit().remove(KEY_PREFIX_SAVE + gameId + "_" + slotKey).apply();
    }

    public void saveProgress(String gameId, String jsonProgress) {
        prefs.edit().putString(KEY_PREFIX_PROGRESS + gameId, jsonProgress).apply();
    }

    public String loadProgress(String gameId) {
        return prefs.getString(KEY_PREFIX_PROGRESS + gameId, null);
    }

    public boolean hasProgress(String gameId) {
        return prefs.contains(KEY_PREFIX_PROGRESS + gameId);
    }

    public void deleteProgress(String gameId) {
        prefs.edit().remove(KEY_PREFIX_PROGRESS + gameId).apply();
    }
}

File: 模块商店/功能模块/游戏/games/src/main/java/com/gamecenter/app/SaveManager.kt
package com.gamecenter.app

import android.content.Context
import android.content.SharedPreferences
import dagger.hilt.android.qualifiers.ApplicationContext
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class SaveManager @Inject constructor(
    @ApplicationContext context: Context
) {

    private val prefs: SharedPreferences =
        context.applicationContext.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)

    fun save(gameId: String, slotKey: String, jsonState: String?) {
        prefs.edit().putString(buildSaveKey(gameId, slotKey), jsonState).apply()
    }

    fun load(gameId: String, slotKey: String): String? {
        return prefs.getString(buildSaveKey(gameId, slotKey), null)
    }

    fun hasSave(gameId: String, slotKey: String): Boolean {
        return prefs.contains(buildSaveKey(gameId, slotKey))
    }

    fun deleteSave(gameId: String, slotKey: String) {
        prefs.edit().remove(buildSaveKey(gameId, slotKey)).apply()
    }

    fun saveProgress(gameId: String, jsonProgress: String?) {
        prefs.edit().putString(buildProgressKey(gameId), jsonProgress).apply()
    }

    fun loadProgress(gameId: String): String? {
        return prefs.getString(buildProgressKey(gameId), null)
    }

    fun hasProgress(gameId: String): Boolean {
        return prefs.contains(buildProgressKey(gameId))
    }

    fun deleteProgress(gameId: String) {
        prefs.edit().remove(buildProgressKey(gameId)).apply()
    }

    private fun buildSaveKey(gameId: String, slotKey: String): String {
        return "${KEY_PREFIX_SAVE}${gameId}_$slotKey"
    }

    private fun buildProgressKey(gameId: String): String {
        return "${KEY_PREFIX_PROGRESS}$gameId"
    }

    companion object {
        private const val PREFS_NAME = "GameMatrix_saves"
        private const val KEY_PREFIX_SAVE = "save_"
        private const val KEY_PREFIX_PROGRESS = "progress_"

        @Volatile
        private var instance: SaveManager? = null

        @JvmStatic
        @Deprecated("推荐通过 Hilt 依赖注入获取实例，避免手动管理单例")
        fun getInstance(context: Context): SaveManager {
            return instance ?: synchronized(SaveManager::class.java) {
                instance ?: SaveManager(context.applicationContext).also { instance = it }
            }
        }
    }

    init {
        instance = this
    }
}
